### PR TITLE
chore: snekfetch 4.0.0 Update

### LINKED ISSUE #2581: [Ratelimits] Snekfetch 4.x breaks ratelimits
**Please describe the problem you are having in as much detail as possible:**

After https://github.com/discordjs/discord.js/commit/46af6e3396e94f795c7f539035a36e733139be07 landed, 429 HTTP errors aren't handled as well as before that commit (for example, https://github.com/discordjs/discord.js/commit/c8ecbe6585a615e77deb5f04bf64a9675e73fb41).

I have tested in two bots, my production bot, with commit https://github.com/discordjs/discord.js/commit/6cde6deceee975d76ea3abf40e87bd6ab04655f7, which handled ratelimits correctly, and my test bot, with commit https://github.com/discordjs/discord.js/commit/5071477172053ef9d274aaab16cec0cfbd25b860, which threw a lot of 429 errors to console:

image

Furthermore, I have checked `this.client.rest.handlers`, and got the following:

```javascript
{ '/gateway':
   RequestHandler {
     manager: [RESTManager],
     client: [Smii],
     handle: [Function: bound sequential],
     limit: NaN,
     resetTime: NaN,
     remaining: NaN,
     queue: [],
     busy: false },
  '/oauth2/applications/@​me':
   RequestHandler {
     manager: [RESTManager],
     client: [Smii],
     handle: [Function: bound sequential],
     limit: NaN,
     resetTime: NaN,
     remaining: NaN,
     queue: [],
     busy: false },
  '/channels/339948984351326208/typing':
   RequestHandler {
     manager: [RESTManager],
     client: [Smii],
     handle: [Function: bound sequential],
     limit: 5,
     resetTime: NaN,
     remaining: 4,
     queue: [],
     busy: true },
  '/channels/339948984351326208/messages':
   RequestHandler {
     manager: [RESTManager],
     client: [Smii],
     handle: [Function: bound sequential],
     limit: 5,
     resetTime: NaN,
     remaining: 4,
     queue: [],
     busy: false },
  '/channels/339948984351326208/messages/:id':
   RequestHandler {
     manager: [RESTManager],
     client: [Smii],
     handle: [Function: bound sequential],
     limit: 5,
     resetTime: NaN,
     remaining: 0,
     queue: [],
     busy: false } }
```

**Downgrading to https://github.com/discordjs/discord.js/commit/c8ecbe6585a615e77deb5f04bf64a9675e73fb41 fixes the issue!**

**Include a reproducible code sample here, if possible:**

Install https://github.com/discordjs/discord.js/commit/46af6e3396e94f795c7f539035a36e733139be07 and make your bot send 6 messages in less than a second (to fill the bucket and make it set busy true).

**Further details:**

- discord.js version: master
- node.js version: 10.3.0
- Operating system: Ubuntu 16.04.3 LTS (Xenial Xerus)
- Priority this issue should have – please be realistic and elaborate if possible: High, I'm not the only one who reproduces this issue, several developers with massive bots have experienced this issue and they're currently getting many errors.

- [ ] I found this issue while running code on a __user account__
- [x] I have also tested the issue on latest master, commit hash: https://github.com/discordjs/discord.js/commit/5071477172053ef9d274aaab16cec0cfbd25b860

### PR DESCRIPTION
**Pull request by dragonfire535**
**Please describe the changes this PR makes and why it should be merged:**
This _should_ fix all of the changes made in snekfetch 4, including fixing the previous ratelimit issue. fixes #2581

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/util/DataResolver.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ class DataResolver {
 
     if (typeof resource === 'string') {
       if (/^https?:\/\//.test(resource)) {
-        return snekfetch.get(resource).then(res => res.body instanceof Buffer ? res.body : Buffer.from(res.text));
+        return snekfetch.get(resource).then(res => res.body instanceof Buffer ? res.body : res.raw);
```
Comment (kyranet) [inline src/util/DataResolver.js]: This check is useless, `res.raw` is always a Buffer instance.

Comment (dragonfire535) [reply src/util/DataResolver.js]: Good catch

**Comment by kyranet** [comment]
Also, is there a reason to use [snekfetch](https://www.npmjs.com/package/snekfetch)@4.0.0 instead of [snekfetch](https://www.npmjs.com/package/snekfetch)@4.0.1, which is the latest npm version, published 2 days ago? To avoid a second PR from @dependabot[bot] after this one.

From what I have seen, 4.0.1 only includes this change, which makes the installation lighter, helping developers download snekfetch itself (and discord.js) faster https://github.com/devsnek/snekfetch/commit/32c5da60a2c0401ee6b66cc4a5caa86c3aa20ec8

**Comment by dragonfire535** [comment]
PR should wait on devsnek/snekfetch#41 before being merged.

**Comment by kyranet** [comment]
[snekfetch](https://www.npmjs.com/package/snekfetch/v/4.0.2)@4.0.2 landed with that fix 👍

**Pull request closed without merging**